cmd/jobsClean.go

```diff
@@ -0,0 +1,132 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-storage-azcopy/common"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	type JobsCleanReq struct {
+		withStatus string
+	}
+
+	commandLineInput := JobsCleanReq{}
+
+	// remove a single job's log and plan file
+	jobsCleanCmd := &cobra.Command{
+		Use:     "clean",
+		Aliases: []string{"cl"},
+		Short:   cleanJobsCmdShortDescription,
+		Long:    cleanJobsCmdLongDescription,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("clean command does not accept arguments")
```

Comment:
Don't really know where to put this, but we should probably have a flag like `--keep-last-x-days` or something of that variety.

My thought is, what if somebody runs azcopy sync daily or whatever and only wants to keep the last 7 days of logs and plan files, in case of a failure?

Reply:
So they could do this:

`azcopy clean --keep-days 7`

I like that idea.  Not sure Ze will have time to include it in the release.  I'd be comfortable with us shipping without it if he's under time pressure.

Reply:
I thought about it too, but wasn't sure whether we should use the last modified time of the job files, or the start time of the job.

I'll leave it for now, and wait to receive some user feedback.